File: app/src/debug/java/com/leaguechampions/utils/UrlUtils.java
package com.leaguechampions.utils;

import android.content.Context;

import com.leaguechampions.data.local.Const;
import com.leaguechampions.data.model.Champion;

import java.util.ArrayList;
import java.util.List;
import java.util.Random;

public final class UrlUtils {

    private UrlUtils() {}

    private static final List<String> champions = new ArrayList<String>(17) {{
            add("Amumu");
            add("Corki");
            add("Garen");
            add("Graves");
            add("JarvanIV");
            add("LeeSin");
            add("Lulu");
            add("MissFortune");
            add("Nasus");
            add("Riven");
            add("Shaco");
            add("Sona");
            add("Teemo");
            add("Tryndamere");
            add("Vladimir");
            add("Zed");
            add("Ziggs");
    }};

    public static String getImageUrl(Context context, Champion champion) {
        String championId = champion.getId();
        if (PrefUtils.isMockMode(context)) {
            if (!champions.contains(championId)) {
                championId = champions.get(new Random().nextInt(champions.size()));
            }
            String filename = "ic_poro_" + championId.toLowerCase() + ".png";
            return (Const.isGlide ? "http" : "mock") + ":///images/" + filename;
        } else {
            return Const.URL_BASE+"cdn/"+champion.getVersion()+"/img/champion/"+championId+".png";
        }
    }
}

File: app/src/debug/java/com/leaguechampions/utils/UrlUtils.kt
package com.leaguechampions.utils

import android.content.Context
import com.leaguechampions.data.local.Const
import java.util.Locale
import java.util.Random

object UrlUtils {

    private val championIds = arrayListOf(
            "Amumu",
            "Corki",
            "Garen",
            "Graves",
            "JarvanIV",
            "LeeSin",
            "Lulu",
            "MissFortune",
            "Nasus",
            "Riven",
            "Shaco",
            "Sona",
            "Teemo",
            "Tryndamere",
            "Vladimir",
            "Zed",
            "Ziggs"
    )

    fun getImageUrl(context: Context, championId: String, version: String): String {
        var id = championId
        return if (PrefUtils.isMockMode(context)) {
            val scheme = if (Const.isGlide) "file:///android_asset/" else "mock:///"
            if (!championIds.contains(id)) {
                id = championIds[Random().nextInt(championIds.size)]
            }

            "${scheme}images/ic_poro_${id.toLowerCase(Locale.getDefault())}.png"
        } else {
            "${Const.URL_BASE}cdn/$version/img/champion/$id.png"
        }
    }
}
